python/pdstools/pega_io/Anonymization.py

```diff
@@ -0,0 +1,314 @@
+
+        Returns
+        -------
+        pl.Expr
+            A Polars expression representing the normalized column.
+
+        Examples
+        --------
+        >>> range = [{"min": 0.0, "max": 100.0}]
+        >>> min_max("age", range)
+        Column "age" normalized using min-max scaling.
+        """
+        if range[0]["min"] == range[0]["max"]:  # pragma: no cover
+            print(f"Column {column_name} only contains one value, so returning 0")
+            return pl.lit(0.0).alias(column_name)
+        return (pl.col(column_name) - pl.lit(range[0]["min"])) / (
+            pl.lit(range[0]["max"] - range[0]["min"])
+        )
+
+    @staticmethod
+    def _infer_types(df: pl.DataFrame):
+        """Infers the types of columns in a DataFrame.
+
+        Parameters
+        ----------
+        df (pl.DataFrame):
+            The DataFrame for which to infer column types.
+
+        Returns
+        -------
+        dict:
+            A dictionary mapping column names to their inferred types.
+            The inferred types can be either "numeric" or "symbolic".
+        """
+        types = dict()
+        for col in df.columns:
+            try:
+                ser = df.get_column(col)
+                try:
+                    ser = ser.replace("", None)
+                except Exception:
+                    pass
+                ser.cast(pl.Float64)
+                types[col] = "numeric"
+            except Exception:
+                types[col] = "symbolic"
+
+        return types
+
+    @staticmethod
+    def chunker(files: List[str], size: int):
+        """Split a list of files into chunks of a specified size.
+
+        Parameters
+        ----------
+        files (List[str]):
+            A list of file names.
+        size (int):
+            The size of each chunk.
+
+        Returns
+        -------
+        generator:
+            A generator that yields chunks of files.
+
+        Examples
+        --------
+            >>> files = ['file1.txt', 'file2.txt', 'file3.txt', 'file4.txt', 'file5.txt']
+            >>> chunks = chunker(files, 2)
+            >>> for chunk in chunks:
+            ...     print(chunk)
+            ['file1.txt', 'file2.txt']
+            ['file3.txt', 'file4.txt']
+            ['file5.txt']
+        """
+
+        return (files[pos : pos + size] for pos in range(0, len(files), size))
+
+    def chunk_to_parquet(self, files: List[str], i) -> str:
+        """Convert a chunk of files to Parquet format.
+
+        Parameters:
+        files (List[str]):
+            List of file paths to be converted.
+        temp_path (str):
+            Path to the temporary directory where the Parquet file will be saved.
+        i:
+            Index of the chunk.
+
+        Returns:
+        str: File path of the converted Parquet file.
+        """
+
+        init_df = pl.concat([pl.read_ndjson(n) for n in files], how="diagonal_relaxed")
```

Comment:
can we use readDSExport in order to be able to accommodate other data formats(parquet, arrow, csv...) too?

Reply:
Good point. Right now I don't think we can easily because we're doing batching based on a glob pattern here, which I don't think readDSExport supports. This is a feature request we've received in the past though (#205), so once we've got this functionality in, we should certainly use it here as well!